src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/evosuite/Runner.kt

```diff
@@ -79,31 +84,45 @@ class Runner(private val projectPath: String, private val projectClassPath: Stri
         log.info("Starting EvoSuite with arguments: $cmdString")
         log.info("Results will be saved to $serializeResultPath")
 
-        Thread {
-            val evoSuiteProcess = GeneralCommandLine(cmd)
-            evoSuiteProcess.charset = Charset.forName("UTF-8")
-            evoSuiteProcess.setWorkDirectory(projectPath)
 
-            val handler: ProcessHandler = OSProcessHandler(evoSuiteProcess)
+        ProgressManager.getInstance()
+            .run(
+                object : Task.Backgroundable(project, "EvoSuite: Generating Tests...") {
 
-            val outputBuilder = StringBuilder()
+                    override fun run(indicator: ProgressIndicator) {
+                        indicator.isIndeterminate = false
+                        val evoSuiteProcess = GeneralCommandLine(cmd)
+                        evoSuiteProcess.charset = Charset.forName("UTF-8")
+                        evoSuiteProcess.setWorkDirectory(projectPath)
+                        val handler: ProcessHandler = OSProcessHandler(evoSuiteProcess)
 
-            // attach process listener for output
-            handler.addProcessListener(object : ProcessAdapter() {
-                override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
-                    val text = event.text
-                    outputBuilder.append(text)
-                    log.info(text)
-                }
-            })
+                        // attach process listener for output
+                        handler.addProcessListener(object : ProcessAdapter() {
+                            override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
+                                val text = event.text
+                                val matcher = Pattern.compile("Progress:[>= ]*(\\d+(?:\\.\\d+)?)%").matcher(text)
```

Comment:
What is this pattern for?

Reply:
It's for extracting the progress % out of evosuite's STDOUT